### ISSUE TITLE
SQLTools errors if comment is first line of query

### ISSUE THREAD

**Issue by Akeboshiwind** [xtdb/crux#3628]
When I put a comment as the first line of the query:
```sql
-- test
SELECT * FROM _txs
```

I get this error: `Release called on client which has already been released to the pool.`

If I put a comment anywhere else, the query works just fine:
```sql
SELECT * -- test
FROM /* test */ _txs
--test
```

Postgres has no issues with the same query.

---

To replicate:
- Install the [SQLTools](https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools) VSCode plugin
- Install the [postgresql driver](https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools-driver-pg) VSCode plugin
- Run a local xtdb `docker run --pull=always -tip 5432: 5432 ghcr.io/xtdb/xtdb-standalone-ea`
- [Add a connection](https://vscode-sqltools.mteixeira.dev/en/home#getting-started) to SQLTools
- Run the first query in an `xxx.sql` file (there will be a button saying "run on active connection")

**Comment by egg-juxt** [comment]
I have seen no differences in pgwire traces in xt when running the comment vs no-comment version.

I have also tried with PostgresJS and node-postgres from my own JS code. No issue found.

**Comment by jarohen** [comment]
lgtm now? guessing resolved as part of #3242
